pkg/frontend/querymiddleware/astmapper/instant_splitting_stats.go

```diff
@@ -19,3 +22,39 @@ func (s *InstantSplitterStats) AddSplitQueries(num int) {
 func (s *InstantSplitterStats) GetSplitQueries() int {
 	return s.splitQueries
 }
+
+// SetNoOpSmallRangeInterval set query as a no operation due to a small range interval.
+func (s *InstantSplitterStats) SetNoOpSmallRangeInterval() {
+	s.noOpSmallIntervalQuery = true
+}
+
+// GetNoOpSmallIntervalQuery returns whether the query is a no operation due to a small range interval.
+func (s *InstantSplitterStats) GetNoOpSmallIntervalQuery() bool {
+	if s.GetSplitQueries() > 0 {
+		s.noOpSmallIntervalQuery = false
```

Comment:
A getter should change the internal value.

_Same comment applies in other getters._

Reply:
Completely agree, it was the best way I found to implement the initial logic. But prefer your suggestion, so addressed the changes in commit: https://github.com/grafana/mimir/pull/2642/commits/11d91d816a1e5b36ec02988e57d151e4865df9ae